UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -39,39 +39,39 @@ public class UnitsNetJsonConverter : JsonConverter
             if (reader.ValueType != null)
                 return reader.Value;
 
-            var obj = TryDeserializeIComparable(reader, serializer);
+            object obj = TryDeserializeIComparable(reader, serializer);
             // A null System.Nullable value or a comparable type was deserialized so return this
             if (!(obj is ValueUnit vu))
                 return obj;
 
             // "MassUnit.Kilogram" => "MassUnit" and "Kilogram"
-            var unitEnumTypeName = vu.Unit.Split('.')[0];
-            var unitEnumValue = vu.Unit.Split('.')[1];
+            string unitEnumTypeName = vu.Unit.Split('.')[0];
+            string unitEnumValue = vu.Unit.Split('.')[1];
 
             // "UnitsNet.Units.MassUnit,UnitsNet"
-            var unitEnumTypeAssemblyQualifiedName = "UnitsNet.Units." + unitEnumTypeName + ",UnitsNet";
+            string unitEnumTypeAssemblyQualifiedName = "UnitsNet.Units." + unitEnumTypeName + ",UnitsNet";
 
             // -- see http://stackoverflow.com/a/6465096/1256096 for details
-            var unitEnumType = Type.GetType(unitEnumTypeAssemblyQualifiedName);
+            Type unitEnumType = Type.GetType(unitEnumTypeAssemblyQualifiedName);
             if (unitEnumType == null)
             {
                 var ex = new UnitsNetException("Unable to find enum type.");
                 ex.Data["type"] = unitEnumTypeAssemblyQualifiedName;
                 throw ex;
             }
 
-            var value = vu.Value;
-            var unitValue = (Enum)Enum.Parse(unitEnumType, unitEnumValue); // Ex: MassUnit.Kilogram
+            double value = vu.Value;
+            Enum unitValue = (Enum)Enum.Parse(unitEnumType, unitEnumValue); // Ex: MassUnit.Kilogram
 
             return Quantity.From(value, unitValue);
         }
 
         private static object TryDeserializeIComparable(JsonReader reader, JsonSerializer serializer)
         {
-            var token = JToken.Load(reader);
+            JToken token = JToken.Load(reader);
```

Comment:
I see you reverted all the types back. That's fine, although I only meant for the few places where the exact type wasn't obvious to have explicit types.